fdbserver/Status.actor.cpp

```diff
@@ -1948,6 +1948,13 @@ static JsonBuilderObject tlogFetcher(int* logFaultTolerance, const std::vector<T
 	int localSetsWithNonNegativeFaultTolerance = 0;
 
 	for (int i = 0; i < tLogs.size(); i++) {
+		if (tLogs[i].tLogs.size() == 0) {
+			// We can have LogSets where there are no tLogs but some LogRouters. Its the way
+			// recruiting is implemented for old LogRouters in TagPartitionedLogSystem, where
+			// it adds an empty LogSet for missing locality.
+			continue;
```

Comment:
Maybe add an assertion here that `tLogs[i].isLocal == false`?

Reply:
My understanding is that you can have LogSet here which would have LogRouters in it, but `isLocal = true`, as recruitment is done from master.

Reply:
Why does it needs to "adds an empty LogSet for missing locality".
Are those LogRouters without locality the ones used to reconstruct txnStateStore and speed-up the recovery?

Reply:
So rephrase, adding LogSet for localities with no tLogs for that locality. https://github.com/apple/foundationdb/blob/7066513109f7b7f4ae716e07c1b595251cafcb18/fdbserver/TagPartitionedLogSystem.actor.cpp#L1986-L1991

I'm not sure why it would need to recruit old LogRouters in localities with no tLogs, but seems harmless. I saw some correctness failures when removing that, but didn't dig into that, as seemed like this is safe way to quickly fix FT for now.